src/dune_rules/file_binding.ml

```diff
@@ -4,26 +4,77 @@ open Memo.O
 type ('src, 'dst) t =
   { src : 'src
   ; dst : 'dst option
+      (* The [dune_syntax] field is used for validation which has different
+         behaviour depending on the version of dune syntax in use. *)
+  ; dune_syntax : Syntax.Version.t
   }
 
-let equal f g { src; dst } t = f src t.src && Option.equal g dst t.dst
+let equal f g { src; dst; dune_syntax } t =
+  f src t.src
+  && Option.equal g dst t.dst
+  && Syntax.Version.equal dune_syntax t.dune_syntax
+;;
+
+let relative_path_starts_with_parent relative_path =
+  match String.lsplit2 relative_path ~on:'/' with
+  | None -> Filename.(equal relative_path parent_dir_name)
+  | Some (first, _) -> String.equal first Filename.parent_dir_name
+;;
+
+let validate_dst_for_install_stanza
+  ~relative_dst_path_starts_with_parent_error_when
+  ~loc
+  dst
+  dune_syntax
+  =
+  if relative_path_starts_with_parent dst
+  then (
+    match relative_dst_path_starts_with_parent_error_when with
+    | `Deprecation_warning_from_3_11 ->
+      let open Syntax.Version.Infix in
+      if dune_syntax >= (3, 11)
+      then
+        User_warning.emit
```

Comment:
Warnings should be disabled in vendored projects.

Reply:
I don't understand this comment. What are you suggesting I change here?

Reply:
In a project that's under a `(vendored_dirs ..)`, we omit all the warnings. This is so that projects that emit projects used by opam-monorepo (or vendored otherwise), do not spam the main projects with warnings.